src/bytes_mut.rs

```diff
@@ -551,6 +551,9 @@ impl BytesMut {
     /// and the original buffer is large enough to fit the requested additional
     /// capacity, then reallocations will never happen.
     ///
+    /// Once this method has been called, the contents of memory outside of
+    /// `0..len` are undefined and may be uninitialized.
```

Comment:
The word undefined is somewhat ambiguous in this context. Let's avoid that. How about this?
```suggestion
    /// This method does not preserve data stored in the unused capacity.
```